src/main/kotlin/com/beust/klaxon/Lookup.kt

```diff
@@ -26,6 +27,11 @@ fun JsonObject.string(fieldName: String) : String? = get(fieldName) as String?
 fun JsonObject.double(fieldName: String) : Double? = get(fieldName) as Double?
 fun JsonObject.boolean(fieldName: String) : Boolean? = get(fieldName) as Boolean?
 
+fun JsonObject.date(fieldName: String, formatter: (String) -> Date?) : Date? {
+    val value: String = get(fieldName) as? String ?: return null
+    return formatter(value)
+}
+
```

Comment:
I think you can simplify as follows:
```
= (get(fieldName) as? String?).apply { formatter(value) }
```

(I haven't tested it, just writing it off the top of my head)